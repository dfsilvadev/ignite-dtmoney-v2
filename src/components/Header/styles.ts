import styled, { css } from "styled-components";

export const HeaderContainer = styled.header`
  ${({ theme }) => css`
    background: ${theme.colors["gray-900"]};
    padding: 2.5rem 0 7.5rem;
  `}
`;

export const HeaderContent = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`;
