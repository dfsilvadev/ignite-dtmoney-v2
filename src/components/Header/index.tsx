import * as S from "./styles";

import Button from "../Button";

import logoSvg from "../../assets/images/svg/logo.svg";
import { Plus } from "phosphor-react";

const Header = () => {
  return (
    <S.HeaderContainer>
      <S.HeaderContent>
        <img
          src={logoSvg}
          alt="Logotipo DT Money"
          title="Logotipo DT Money"
          aria-label="Logotipo DT Money"
        />

        <Button type="button" size="small" icon={<Plus weight="bold" />}>
          Nova Transação
        </Button>
      </S.HeaderContent>
    </S.HeaderContainer>
  );
};

export default Header;
