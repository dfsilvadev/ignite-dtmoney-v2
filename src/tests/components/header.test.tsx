import { screen } from "@testing-library/react";
import { renderWithTheme } from "../../utils/tests/renderWithTheme";

import Header from "../../components/Header";

describe("Header", () => {
  it("should render the heading", () => {
    const { container } = renderWithTheme(<Header />);

    expect(
      screen.getByRole("img", { name: /logotipo dt money/i })
    ).toBeInTheDocument();

    expect(screen.getByRole("button")).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
