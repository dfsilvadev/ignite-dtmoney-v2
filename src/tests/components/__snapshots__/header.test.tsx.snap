// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render the heading 1`] = `
.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: #00875F;
  color: #fff;
  border: 0;
  border-radius: 0.25rem;
  padding: 1rem;
  height: 2.5rem;
  font-size: 0.875rem;
}

.c3:hover {
  background: #015F43;
}

.c3 svg {
  width: 1.5rem;
}

.c3 svg + span {
  margin-left: 0.5rem;
}

.c1 {
  width: 100%;
  max-width: 70rem;
  margin: 0 auto;
  padding: 0 1rem;
}

.c0 {
  background: #121214;
  padding: 2.5rem 0 7.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

<div>
  <header
    class="c0"
  >
    <header
      class="c1"
    >
      <div
        class="c2"
      >
        <img
          alt="Logotipo DT Money"
          aria-label="Logotipo DT Money"
          src="SvgrURL"
          title="Logotipo DT Money"
        />
        <button
          aria-controls="radix-:r0:"
          aria-expanded="false"
          aria-haspopup="dialog"
          aria-label="new transaction"
          class="c3"
          data-state="closed"
          type="button"
        >
          <svg
            fill="currentColor"
            height="1em"
            viewBox="0 0 256 256"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <rect
              fill="none"
              height="256"
              width="256"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="24"
              x1="40"
              x2="216"
              y1="128"
              y2="128"
            />
            <line
              fill="none"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="24"
              x1="128"
              x2="128"
              y1="40"
              y2="216"
            />
          </svg>
          <span>
            Nova Transação
          </span>
        </button>
      </div>
    </header>
  </header>
</div>
`;
